### PR TITLE
[DEV-3321] Refetch Verification Status

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

> Refetches verification status on a timer and automatically redirects the user when found to be verified.
> Will also now redirect users that arrive on `/verify` that have already been verified instead of erroring.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links
- Coda issue number: https://coda.io/d/PRJ-223-Refetch-Microsite-Verification-Status_dDh2WfzORSi/Scope_sucSR
- Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3321
- Process.st launch checklist: https://app.process.st/runs/DEV-3321-Refetch-Email-Verification-Status-oVwNWCAuswsvBIaDTJNF9g

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run (if applicable)
- [x] The behaviour changes in the pull request are covered by specs
- [x] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist

### Code review

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: packages/mint-components/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.6.3] - 2022-10-03
```
Comment (DerekSiemens) [inline packages/mint-components/CHANGELOG.md:10]: Flagging date bump required

File: packages/mint-components/src/index.html [hunk 2]
```diff
@@ -10,8 +10,201 @@
...
+          slot="footer"
+          support-email="support@example.com"
+          terms-link="https://example.com"
+          faq-link="https://example.com"
+          terms-text="Terms And Conditions"
+          faq-text="FAQ"
+          show-powered-by="true"
+        ></sqm-portal-footer>
+      </sqm-portal-frame>
+      </sqm-brand>
```
Comment (Locrian24) [inline packages/mint-components/src/index.html]: Should go back to stencilbook component

File: packages/mint-components/src/components/sqm-portal-email-verification/usePortalEmailVerification.tsx [hunk 3]
```diff
@@ -1,30 +1,50 @@
...
-  const [success, setSuccess] = useState(false);
-  const [error, setError] = useState("");
   const urlParams = new URLSearchParams(navigation.location.search);
   const nextPage = urlParams.get("nextPage");
 
+  const [success, setSuccess] = useState(false);
+  const [error, setError] = useState("");
+  const [countdown, setCountdown] = useState(10);
+  const timerRef = useRef<any>(undefined);
+  const countdownRef = useRef<any>(undefined);
```
Comment (Locrian24) [inline packages/mint-components/src/components/sqm-portal-email-verification/usePortalEmailVerification.tsx:39]: Flagging `any` types

File: packages/mint-components/src/components/sqm-portal-email-verification/usePortalEmailVerification.tsx [hunk 4]
```diff
@@ -34,6 +54,66 @@ export function usePortalEmailVerification(props: PortalEmailVerification) {
...
+      timerRef.current = startTimer();
+    }
+    return () => {
+      clearInterval(timerRef.current);
+    };
+  }, []);
+
+  // Countdown timer
+  useEffect(() => {
+    if (countdown > 0) countdownRef.current = startCountdown(countdown - 1);
```
Comment (Locrian24) [inline packages/mint-components/src/components/sqm-portal-email-verification/usePortalEmailVerification.tsx:108]: Is this safe? There's a new interval every time the countdown ticks but it'll only clean up the last one

**Review by Locrian24 — COMMENTED** [review]
Minor thing with all intervals not being explicitly cleared. Not sure if this actually causes a memory leak but worth confirming

**Comment by DerekSiemens** [comment]
We likely need a quick extra 2 stories added, one for during the countdown and one when its loading 
image

**Comment by DerekSiemens** [comment]
Also the current stories are a bit broken with the new prop, small fix up required 
image

**Comment by noahwc** [comment]
@DerekSiemens stories updated to fix message and include countdown/loading states for verification refetch.

File: packages/mint-components/CHANGELOG.md (hunk 1 above)
Comment (DerekSiemens) [inline packages/mint-components/CHANGELOG.md:10]: Flagging change log update

File: packages/mint-components/src/components/sqm-portal-email-verification/usePortalEmailVerification.tsx (hunk 4 above)
Comment (00salmon) [reply packages/mint-components/src/components/sqm-portal-email-verification/usePortalEmailVerification.tsx]: Not 100% sure, but it now clears the interval each count just to be safe.

**Review by Locrian24 — APPROVED** [review]

**Pull request merged**